pkg/deploy/api/v1beta1/types.go

```diff
@@ -180,6 +180,12 @@ const (
 	// DeploymentConfigLabel is the name of a label used to correlate a deployment with the
 	// DeploymentConfigs on which the deployment is based.
 	DeploymentConfigLabel = "deploymentconfig"
+	// DeploymentStatusReasonAnnotation represents the reason for deployment being in a given state
+	// Used for specifying the reason for cancellation or failure of a deployment
+	DeploymentStatusReasonAnnotation = "deploymentStatusReason"
+	// DeploymentCancelledAnnotation indicates that the deployment has been cancelled
+	// The annotation value does not matter and its mere presence indicates cancellation
+	DeploymentCancelledAnnotation = "deploymentCancelled"
```

Comment:
All new annotations should be of the new form - we should not add any older annotations.  Annotations do not correspond to an API version, instead, they are an evolving schema with implicit versions